frontend/src/components/nodes/nodeSidebar/NodeSidebar.tsx

```diff
@@ -149,17 +93,19 @@ const NodeSidebar: React.FC<NodeSidebarProps> = ({ nodeID }) => {
   const nodeTypesMetadata = useSelector((state: RootState) => state.nodeTypes as NodeType).metadata;
   const node = useSelector((state: RootState) => selectNodeById(state, nodeID));
   const storedWidth = useSelector((state: RootState) => state.flow.sidebarWidth);
+  const nodeConfig = useSelector((state: RootState) => state.flow.nodeConfigs[nodeID]);
+  const allNodeConfigs = useSelector((state: RootState) => state.flow.nodeConfigs);
 
   const hasRunOutput = !!node?.data?.run;
 
   const [width, setWidth] = useState<number>(storedWidth);
   const [isResizing, setIsResizing] = useState<boolean>(false);
 
   const [nodeType, setNodeType] = useState<string>(node?.type || 'ExampleNode');
-  const [nodeSchema, setNodeSchema] = useState<NodeSchema | null>(
+  const [nodeSchema, setNodeSchema] = useState<NodeType | null>(
     findNodeSchema(node?.type || 'ExampleNode', nodeTypes)
   );
-  const [dynamicModel, setDynamicModel] = useState<DynamicModel>(node?.data?.config || {});
+  const [dynamicModel, setDynamicModel] = useState<FlowWorkflowNodeConfig>(nodeConfig || {});
```

Comment:
I like that we have defined a type for the config information: `FlowWorkflowNodeConfig`.
Can we change the name of the `dynamicModel` state to something more apt as well? Something like `localNodeConfig` or `draftNodeConfig` to reflect the function of this variable.